api/v1/cluster_types.go

```diff
@@ -1037,6 +1037,9 @@ type S3Credentials struct {
 	// The reference to the secret access key
 	SecretAccessKeyReference *SecretKeySelector `json:"secretAccessKey,omitempty"`
 
+	// The reference to the secret containing the region name
+	RegionReference *SecretKeySelector `json:"region,omitempty"`
+
```

Comment:
I think it's missing a `make generate`, I do not see anything in the deepcopy file related to this